### PR TITLE
8266286: Add LoadVectorGather and StoreVectorScatter's partial version

### PR DESCRIPTION
**Pull request by Wanghuang-Huawei**
* Add LoadVectorGather and StoreVectorScatter's partial version
*  use `sve_whilelo`

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8266286](https://bugs.openjdk.java.net/browse/JDK-8266286): Add LoadVectorGather and StoreVectorScatter's partial version

### Reviewers
 * [Xiaohong Gong](https://openjdk.java.net/census#xgong) (@XiaohongGong - Committer) ⚠️ Review applies to dd90996208c15ee7954908ef8c5c0561903ed6f5
 * [Ningsheng Jian](https://openjdk.java.net/census#njian) (@nsjian - Committer)

### Contributors
 * Wang Huang `<whuang@openjdk.org>`
 * Ai Jiaming `<aijiaming1@huawei.com>`

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-vector pull/88/head:pull/88` \
`$ git checkout pull/88`

Update a local copy of the PR: \
`$ git checkout pull/88` \
`$ git pull https://git.openjdk.java.net/panama-vector pull/88/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 88`

View PR using the GUI difftool: \
`$ git pr show -t 88`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-vector/pull/88.diff">https://git.openjdk.java.net/panama-vector/pull/88.diff</a>

</details>

### CODE REVIEW

**Review by XiaohongGong — APPROVED** [review]
Tests pass with SVE 512 vector length. So looks good to me. Thanks!

File: src/hotspot/cpu/aarch64/aarch64_sve.ad [hunk 1]
```diff
@@ -4167,6 +4167,90 @@ instruct scatterL(indirect mem, vReg src, vReg idx) %{
   ins_pipe(pipe_slow);
 %}
 
+// ------------------------------ Vector Load Gather Partial-------------------------------
+instruct gatherI_partial(vReg dst, indirect mem, vReg idx, pRegGov pTmp, rFlagsReg cr) %{
+  predicate(UseSVE > 0 &&
+            n->as_LoadVectorGather()->memory_size() < MaxVectorSize &&
+           (n->bottom_type()->is_vect()->element_basic_type() == T_INT ||
+            n->bottom_type()->is_vect()->element_basic_type() == T_FLOAT));
```
Comment (nsjian) [inline src/hotspot/cpu/aarch64/aarch64_sve.ad:4175]: Style nit: add one more space at the beginning of these lines to have better indentation?

**Review by nsjian — APPROVED** [review]
Looks good! Thanks!

File: src/hotspot/cpu/aarch64/aarch64_sve.ad (hunk 1 above)
Comment (nsjian) [reply src/hotspot/cpu/aarch64/aarch64_sve.ad:4175]: > Style nit: add one more space at the beginning of these lines to have better indentation?

One more nit: I would suggest to group those gather partial rules together with other gather rules, as well as scatter parts.

Comment (Wanghuang-Huawei) [reply src/hotspot/cpu/aarch64/aarch64_sve.ad]: Sure. I will change that.

**Review by nsjian — APPROVED** [review]

**Pull request closed without merging**